modules/position/src/main/native/android/dalvik/DalvikPositionService.java

```diff
@@ -0,0 +1,326 @@
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL GLUON BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.gluonhq.helloandroid;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.location.Criteria;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.provider.Settings;
+import java.io.*;
+import java.util.*;
+import java.util.logging.*;
+import android.util.Log;
+
+
+/**
+ * An implementation of the
+ * {@link PositionService PositionService} for the
+ * Android platform.
+ * 
+ * Requires ACCESS_COARSE_LOCATION and/or ACCESS_FINE_LOCATION permission:
+ * {@code <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
+ *  <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
+ * 
+ */
+
+public class DalvikPositionService implements LocationListener {
+
+    private static final String TAG = "GluonAttach";
+    private Activity activityContext;
+public static final Parameters DEFAULT_PARAMETERS = new Parameters(Parameters.Accuracy.MEDIUM, false);
+
+    private static final Logger LOG = Logger.getLogger(DalvikPositionService.class.getName());
+    
+    public static final String LONGITUDE = "longitude";
+    public static final String LATITUDE = "latitude";
+    public static final String ALTITUDE = "altitude";
+
+    private EarthGravitationalModel gh;
+    private LocationManager locationManager;
+    private String locationProvider;
+
+    private AndroidLooperTask looperTask = null;
+    private Parameters parameters;
+    private boolean running;
+    private boolean debug = true;
+
+    public DalvikPositionService(Activity activity) {
+        Log.v(TAG, "Construct DalvikPositionService");
+        this.activityContext = activity;
+        boolean gpsEnabled = Util.verifyPermissions(Manifest.permission.ACCESS_COARSE_LOCATION) || 
+                Util.verifyPermissions(Manifest.permission.ACCESS_FINE_LOCATION);
+        if (!gpsEnabled) {
+            Log.v(TAG, "GPS disabled. ACCESS_COARSE_LOCATION or ACCESS_FINE_LOCATION permissions are required");
+        }
+        gh = new EarthGravitationalModel();
+        try {
+            gh.load("egm180.nor");
```

Comment:
The file `egm180.not` is missing from this PR. It should be added to resources, so it is bundled by position:android.jar. However we don't have android/resources, so we'll need some tweak in our build, to avoid adding it to the other jars.

Reply:
this is a follow-up issue: https://github.com/gluonhq/attach/issues/79